libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Maxell.cpp

```diff
@@ -85,6 +126,8 @@ void AP_BattMonitor_SMBus_Maxell::timer()
     read_temp();
 
     read_serial_number();
+
+    read_cycle_count();
```

Comment:
Heh, I can't knock it, because I didn't fix it in the serial number case, but we really should be lowering the frequency at which we do this. SMBus transactions have a much longer timeout then I2C transactions, and the battery is allowed to wander off for a bit, for both the serial number and cycle count we really aren't benefitting by updating these at 10Hz. (SMBus is allowed upto 35 milliseconds per transaction, which gets to be quite a long time to hold the bus for each transaction if the battery isn't responding). Tridge and I discussed some ways to make this less painful in the general case, but we obviously haven't gotten to it yet.

Anyways, this was a long way of saying ideally we'd only update the cycle count here if we didn't have it yet, or it's been 10 seconds since the last update (just in case you were charging in the vehicle). And if we had that logic the serial number should be inside that same protection. But given that we haven't fully done it yet I don't think it's a hard blocker, but it's important to note that this is starting to potentially hold the I2C bus for an impactful amount of time if you had say a compass on the same bus (or airspeed sensor)

Reply:
@WickedShell, both the read-serial-number (pre-existing) and read-cycle-count only attempt to read from the I2C bus once and once they have the value they'll stop.  Also it seems there's 5 second timeout so if it hasn't managed to read the voltage (because the battery is powered down) then it won't be running all these secondary values (temp, serial number and cycle count).

Reply:
Oh yeah, sorry, a bit tired here. I forgot I put that in rather then rate limit it. Be consistent for now is fine, I should go back and fix that though, because I know there are a couple of people looking at hot swapping/in situ charging which would need those to be read again (if at low rate). Anyways not a problem here, sorry for the noise!